torch/_tensor_str.py

```diff
@@ -206,6 +206,8 @@ def _tensor_str(self, indent):
     summarize = self.numel() > PRINT_OPTS.threshold
     if self.dtype is torch.float16 or self.dtype is torch.bfloat16:
         self = self.float()
+    if self.is_hammerblade:
+        self = self.cpu()
```

Comment:
Smart! =) How does cuda do it? `.float()`?

Reply:
Well I don't really know about the cuda way actually. We don't have cuda installed on our server so it won't compile XD